internal/core/command/controller/messaging/external/external.go

```diff
@@ -124,6 +134,83 @@ func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Co
+		topicLevels := strings.Split(message.Topic(), "/")
+		length := len(topicLevels)
+		if length <= 3 {
+			lc.Error("Failed to parse and construct response topic scheme, expected request topic scheme: '#/<device-name>/<command-name>/<method>")
+			lc.Warn("Not publishing error message back due to insufficient information on response topic")
+			return
+		}
+		deviceName := topicLevels[length-3]
+		commandName := topicLevels[length-2]
+		method := topicLevels[length-1]
+		if !strings.EqualFold(method, "get") && !strings.EqualFold(method, "set") {
+			lc.Errorf("Unknown request method: %s, only 'get' or 'set' is allowed", method)
+			lc.Warn("Not publishing error message back due to insufficient information on response topic")
+			return
+		}
+		// expected command response topic scheme: #/<device>/<command-name>/<method>
+		externalResponseTopic := strings.Join([]string{messageBusInfo.External.Topics[ResponseCommandTopicPrefix], deviceName, commandName, method}, "/")
+
+		// retrieve device information through Metadata DeviceClient
+		dc := bootstrapContainer.DeviceClientFrom(dic.Get)
+		if dc == nil {
+			responseEnvelope := types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, "nil Device Client")
+			publishMessage(client, externalResponseTopic, qos, retain, responseEnvelope, lc)
+			return
+		}
+		deviceResponse, err := dc.DeviceByName(context.Background(), deviceName)
+		if err != nil {
+			errorMessage := fmt.Sprintf("Failed to get Device by name %s: %v", deviceName, err)
+			responseEnvelope := types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, errorMessage)
+			publishMessage(client, externalResponseTopic, qos, retain, responseEnvelope, lc)
+			return
+		}
+
+		// retrieve device service information through Metadata DeviceClient
+		dsc := bootstrapContainer.DeviceServiceClientFrom(dic.Get)
+		if dsc == nil {
+			responseEnvelope := types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, "nil DeviceService Client")
+			publishMessage(client, externalResponseTopic, qos, retain, responseEnvelope, lc)
+			return
+		}
+		deviceServiceResponse, err := dsc.DeviceServiceByName(context.Background(), deviceResponse.Device.ServiceName)
+		if err != nil {
+			errorMessage := fmt.Sprintf("Failed to get DeviceService by name %s: %v", deviceResponse.Device.ServiceName, err)
+			responseEnvelope := types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, errorMessage)
+			publishMessage(client, externalResponseTopic, qos, retain, responseEnvelope, lc)
+			return
+		}
+
+		// expected internal command request topic scheme: #/<device-service>/<device>/<command-name>/<method>
+		internalRequestTopic := strings.Join([]string{messageBusInfo.Internal.Topics[RequestTopicPrefix], deviceServiceResponse.Service.Name, deviceName, commandName, method}, "/")
+		internalMessageBus := bootstrapContainer.MessagingClientFrom(dic.Get)
+		requestEnvelope.ReceivedTopic = internalRequestTopic
```

Comment:
This is set by implementations in go-mod-messaging when message is received. No need to set it here since you are publishing, not receiving...

Reply:
reverted the commit about this change